### PR TITLE
Slam 29 - Added executable to programmatically navigate to point on map

### PR DESCRIPTION
**Pull request by Federico-PizarroBejarano**
Modified movebase_client.py to take in frame, added new executable the at sends goals to map by calling movebase_client.py by passing 'map' as frameid. Old executable passes 'base_link' as frameid. The navigation on map works when the move_base is working on the map frame (amcl, gmapping) but not when its just rolling window. The base_link navigation always works.

### CODE REVIEW

File: slam-ws/src/nav_goal/bin/movebase_client_map [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            x = float(sys.argv[1])
+            y = float(sys.argv[2])
+            yaw = float(sys.argv[3])
+        except:
+            rospy.logerr("Error: Incorrect Argument Types - All 3 arguments must be floats or integers.")
+            rospy.signal_shutdown("Error: Incorrect Argument Types - All 3 arguments must be floats or integers.")
+            sys.exit(1)
+
+       	result = nav_goal.movebase_client.movebase_client(x, y, yaw, "map")
+       	rospy.loginfo(result)
```
Comment (erickmu1) [inline slam-ws/src/nav_goal/bin/movebase_client_map]: The indentation seems wrong here. Also it is different `bin/movebase_client`:
```
try:
    result = nav_goal.movebase_client.movebase_client(x, y, yaw, "base_link")
    rospy.loginfo(result)
except:
    rospy.logerr("Error - Navigation failed for unknown reasons. Exiting...")
    rospy.signal_shutdown("Error - Navigation failed for unknown reasons. Exiting...")
```

Comment (erickmu1) [inline slam-ws/src/nav_goal/bin/movebase_client_map]: In general, this code is a lot of copy paste from`bin/movebase_client`. Can we just call `bin/movebase_client` from `bin/movebase_client_map` and just specify input parameters? If you can't call it then I would suggest turning the whole argument parsing and calling `movebase_client()` into a function in `movebase_client.py` and just call this function in each of the executable's `__main__`

Comment (Federico-PizarroBejarano) [reply slam-ws/src/nav_goal/bin/movebase_client_map]: The indentation is 4 spaces everywhere I think

Comment (Federico-PizarroBejarano) [reply slam-ws/src/nav_goal/bin/movebase_client_map]: Done

File: slam-ws/src/nav_goal/src/nav_goal/init_node.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            x = float(arguments[1])
+            y = float(arguments[2])
+            yaw = float(arguments[3])
+
+            return (x, y, yaw)
+        except:
+            rospy.logerr("Error: Incorrect Argument Types - All 3 arguments must be floats or integers.")
+            rospy.signal_shutdown("Error: Incorrect Argument Types - All 3 arguments must be floats or integers.")
+            return 'Failure'
+
```
Comment (erickmu1) [inline slam-ws/src/nav_goal/src/nav_goal/init_node.py]: Can you just put this in `movebase_client.py` as another function, I don't believe we need another file for this

Comment (Federico-PizarroBejarano) [reply slam-ws/src/nav_goal/src/nav_goal/init_node.py]: Done

**Pull request merged**